Documentation/colors/literals-swift3.md

```diff
@@ -0,0 +1,45 @@
+## Template Information
+
+| Name      | Description       |
+| --------- | ----------------- |
+| File name | colors/literals-swift3.stencil |
+| Invocation example | `swiftgen colors -t literals-swift3 …` |
+| Language | Swift 3 |
+| Author | Olivier Halligon |
+
+## When to use it
+
+- When you need to generate *Swift 3* code
+- Supports _multiple_ color names with the _same_ value
+- Uses `#colorLiteral`s for easy preview of the actual color
+
+## Customization
+
+You can customize some elements of this template by overriding the following parameters when invoking `swiftgen` in the command line, using `--param <paramName>=<newValue>`
+
+| Parameter Name | Default Value | Description |
+| -------------- | ------------- | ----------- |
+| `enumName` | `ColorName` | Allows you to change the name of the generated `enum` containing all colors. |
```

Comment:
Add a note (same in the documentations for other `literals-swiftN` of course) about the special `Color` name (maybe outside the table as it can be a paragraph of its own), e.g.

> Note: if you use `--param enumName=Color` then the color constants will be generated as an extension of the `UIColor` (iOS) / `NSColor` (macOS) type directly without creating a separate `enum` type for namespacing those color constants

But tbf I think it would be less surprising if the test was on testing `UIColor` or `NSColor`, not `Color`.

Like adapting the template to:

```
{% set enumName %}{{param.enumName|default:"ColorName"}}{% endset %}
#if os(OSX)
  import AppKit.NSColor
  typealias Color = NSColor
  {% if enumName != 'NSColor' %}enum {{enumName}} { }{% endif %}
#elseif os(iOS) || os(tvOS) || os(watchOS)
  import UIKit.UIColor
  typealias Color = UIColor
  {% if enumName != 'UIColor' %}enum {{enumName}} { }{% endif %}
#endif

// swiftlint:disable file_length identifier_name line_length type_body_length
extension {{enumName}} {
…
}
```
And then explaining that in the doc that the exception is with `--param enumName=UIColor` for those who wish to make those constants as an extension of `UIColor` directly. Would feel more natural as a special name imho?